Find a functionally equivalent code:android.media.AudioRecord.audioParamCheck:COMMENT
Method Modifier: private     
Comment:// mSampleRate is valid

@@ -9,35 +9,14 @@
         throw new IllegalArgumentException(sampleRateInHz + "Hz is not a supported sample rate.");
     }
     mSampleRate = sampleRateInHz;
-    // channel config
-    switch(channelConfig) {
-        // AudioFormat.CHANNEL_CONFIGURATION_DEFAULT
-        case AudioFormat.CHANNEL_IN_DEFAULT:
-        case AudioFormat.CHANNEL_IN_MONO:
-        case AudioFormat.CHANNEL_CONFIGURATION_MONO:
-            mChannelCount = 1;
-            mChannelMask = AudioFormat.CHANNEL_IN_MONO;
-            break;
-        case AudioFormat.CHANNEL_IN_STEREO:
-        case AudioFormat.CHANNEL_CONFIGURATION_STEREO:
-            mChannelCount = 2;
-            mChannelMask = AudioFormat.CHANNEL_IN_STEREO;
-            break;
-        case (AudioFormat.CHANNEL_IN_FRONT | AudioFormat.CHANNEL_IN_BACK):
-            mChannelCount = 2;
-            mChannelMask = channelConfig;
-            break;
-        default:
-            throw new IllegalArgumentException("Unsupported channel configuration.");
-    }
     // audio format
     switch(audioFormat) {
         case AudioFormat.ENCODING_DEFAULT:

